emperor/support_files/js/emperor.js

```diff
@@ -125,137 +132,136 @@ function toggleScaleCoordinates(element){
+		operation(g_sceneLight.position.y, ax0_explained),
+		operation(g_sceneLight.position.z, ax0_explained)
   );
 
 	// scale the position of the camera according to pc1
 	g_sceneCamera.position.set(
-		operation(g_sceneCamera.position.x, g_fractionExplained[g_viewingAxes[0]]),
-		operation(g_sceneCamera.position.y, g_fractionExplained[g_viewingAxes[0]]),
-		operation(g_sceneCamera.position.z, g_fractionExplained[g_viewingAxes[0]])
+		operation(g_sceneCamera.position.x, ax0_explained),
+		operation(g_sceneCamera.position.y, ax0_explained),
+		operation(g_sceneCamera.position.z, ax0_explained)
   );
 	// scale the axis lines
 	drawAxisLines();
 
 	// set the new position of each of the sphere objects
-  _.each(g_plotSpheres, function(sphere){
-    // scale the position of the spheres
-    sphere.position.set(
-			operation(sphere.position.x, g_fractionExplained[g_viewingAxes[0]]),
-			operation(sphere.position.y, g_fractionExplained[g_viewingAxes[1]]),
-			operation(sphere.position.z, g_fractionExplained[g_viewingAxes[2]])
-    );
-  });
+    for(i = 0; i < g_plotSpheres.length; i++) {
+        // scale the position of the spheres
+        pos = g_plotSpheres[i].position;
+        pos.set(operation(pos.x, ax0_explained),
+                operation(pos.y, ax1_explained),
+                operation(pos.z, ax2_explained)
+        );
+    }
 
 	// ellipses won't always be available hence the two separate loops
-  _.each(g_plotEllipses, function(ellipse){
+    for(i = 0; i < g_plotEllipses.length; i++) {
   		// scale the dimensions of the positions of each ellipse
-      ellipse.position.set(
-  			operation(ellipse.position.x, g_fractionExplained[g_viewingAxes[0]]),
-  			operation(ellipse.position.y, g_fractionExplained[g_viewingAxes[1]]),
-  			operation(ellipse.position.z, g_fractionExplained[g_viewingAxes[2]])
-      );
+        ellipsis = g_plotEllipses[i];
+        pos = ellipsis.position;
+        scale = ellipsis.scale;
+
+        pos.set(operation(pos.x, ax0_explained),
+  			    operation(pos.y, ax1_explained),
+  			    operation(pos.z, ax2_explained)
+        );
 
   		// scale the dimensions of the ellipse
-      ellipse.scale.set(
-  			operation(ellipse.scale.x, g_fractionExplained[g_viewingAxes[0]]),
-  			operation(ellipse.scale.y, g_fractionExplained[g_viewingAxes[1]]),
-  			operation(ellipse.scale.z, g_fractionExplained[g_viewingAxes[2]])
+        scale.set(operation(scale.x, ax0_explained),
+  			      operation(scale.y, ax1_explained),
+  			      operation(scale.z, ax2_explained)
       );
-  });
+    }
 
-  _.each(g_plotTaxa, function (taxa){
-    //scale the dimensions of the positions of each taxa-sphere
-		taxa.position.set(
-			operation(taxa.position.x, g_fractionExplained[g_viewingAxes[0]]),
-			operation(taxa.position.y, g_fractionExplained[g_viewingAxes[1]]),
-			operation(taxa.position.z, g_fractionExplained[g_viewingAxes[2]])
-    );
+    for(i = 0; i < g_plotTaxa.length; i++) {
```

Comment:
I'm way late to this party, but doing this won't work, mainly because g_plotTaxa is not an `Array`, but instead an object, it seems like we have two solutions moving forward. We can either change these variables to be arrays instead of objects (g_plotTaxa, g_plotSpheres, etc), this way we can manipulate the points the way we want or we revert back to using `_.each`